crates/wast/src/component/types.rs

```diff
@@ -431,8 +436,8 @@ impl<'a> Parse<'a> for VariantCase<'a> {
 /// A refinement for a variant case.
 #[derive(Debug)]
 pub enum Refinement<'a> {
-    /// The refinement is named.
-    Named(Span, &'a str),
+    /// The refinement is referenced by identifier.
+    Id(Span, Id<'a>),
```

Comment:
I think this will want to be `Index<'a>` instead of `Id<'a>` to accept integers in addition to `$foo` (like other identifiers in the text format), which would also remove the need for wasmprinter to inject `$cNN` and simply print the number for now

Reply:
I'll fix; I was hoping we might force users to refer to the case by id to avoid issues introduced by order changes, but if you're using `wat`, you're no stranger to that methinks.